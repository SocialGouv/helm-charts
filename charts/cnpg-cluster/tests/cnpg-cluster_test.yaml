suite: test cnpg-cluster
templates:
  - cluster.cnpg.yaml
  - pooler.cnpg.yaml
  - backup-cron.yaml
  - scheduledbackup.cnpg.yaml
tests:
  - it: cluster should render
    asserts:
      - template: cluster.cnpg.yaml
        isKind:
          of: Cluster
      - template: cluster.cnpg.yaml
        hasDocuments:
          count: 1
      - template: backup-cron.yaml
        hasDocuments:
          count: 0
      # waiting for release of https://github.com/helm-unittest/helm-unittest/commit/0ace2cc039c1fa33133ea1f26e7cae620443d42a
      # - containsDocument:
      #     kind: CronJob
      #   not: true
  - it: cluster with custom instances
    values:
      - ./values/instances.yaml
    asserts:
      - template: cluster.cnpg.yaml
        isKind:
          of: Cluster
      - template: cluster.cnpg.yaml
        equal:
          path: spec.instances
          value: 2
  - it: cluster with custom image tag
    values:
      - ./values/tag.yaml
    asserts:
      - template: cluster.cnpg.yaml
        isKind:
          of: Cluster
      - template: cluster.cnpg.yaml
        equal:
          path: spec.imageName
          value: ghcr.io/cloudnative-pg/postgis:12
  - it: cluster with scheduled backup enabled
    values:
      - ./values/backup.yaml
    asserts:
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec
      - template: scheduledbackup.cnpg.yaml
        equal:
          path: spec.schedule
          value: "1 2 3 * * 0"
      - template: backup-cron.yaml
        equal:
          path: spec.schedule
          value: "4 5 * * 0"
      - template: backup-cron.yaml
        hasDocuments:
          count: 1
  - it: cluster with recovery enabled
    values:
      - ./values/recovery.yaml
    asserts:
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec
  - it: cluster with enabled backup and recovery
    values:
      - ./values/recovery.yaml
      - ./values/backup.yaml
    asserts:
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec
      - template: backup-cron.yaml
        hasDocuments:
          count: 1
  - it: cluster with custom pgparams
    values:
      - ./values/pgparams.yaml
    asserts:
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec.postgresql
  - it: cluster with custom pg_hba
    values:
      - ./values/pg_hba.yaml
    asserts:
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec.postgresql
  - it: cluster with replication
    values:
      - ./values/replication.yaml
    asserts:
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec.externalClusters
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec.bootstrap
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec.replica
  - it: cluster with pg_basebackup
    values:
      - ./values/pg_basebackup.yaml
    asserts:
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec.externalClusters
      - template: cluster.cnpg.yaml
        matchSnapshot:
          path: spec.bootstrap
  - it: cluster with monitoring
    values:
      - ./values/monitor.yaml
    asserts:
      - template: pooler.cnpg.yaml
        matchSnapshot: {}
