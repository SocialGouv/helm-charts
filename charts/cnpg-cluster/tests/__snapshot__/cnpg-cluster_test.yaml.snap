cluster with custom pgparams:
  1: |
    parameters:
      timezone: Europe/Paris
      work_mem: 512MB
cluster with enabled backup and recovery:
  1: |
    concurrencyPolicy: Forbid
    jobTemplate:
      spec:
        backoffLimit: 0
        template:
          metadata:
            labels:
              app: cnpg-backup-s3-client
            name: backup-cron
          spec:
            containers:
              - env:
                  - name: AWS_ACCESS_KEY_ID
                    valueFrom:
                      secretKeyRef:
                        key: bucket_access_key
                        name: minio
                  - name: AWS_SECRET_ACCESS_KEY
                    valueFrom:
                      secretKeyRef:
                        key: bucket_secret_key
                        name: minio
                  - name: AWS_DEFAULT_REGION
                    valueFrom:
                      secretKeyRef:
                        key: bucket_region
                        name: minio
                  - name: AWS_ENDPOINT_URL
                    value: http://minio:9000
                  - name: DESTINATION_PATH
                    value: s3://backups/dumps
                envFrom:
                  - secretRef:
                      name: pg-hasura-app
                image: ghcr.io/socialgouv/docker/s3-client:1
                imagePullPolicy: IfNotPresent
                name: s3-client
                securityContext:
                  allowPrivilegeEscalation: false
            restartPolicy: Never
            securityContext:
              fsGroup: 1001
              runAsGroup: 1001
              runAsUser: 1001
    schedule: 4 5 6 * * 0
  2: |
    backup:
      barmanObjectStore:
        destinationPath: s3://backups/
        endpointURL: http://minio:9000
        s3Credentials:
          accessKeyId:
            key: ACCESS_KEY_ID
            name: minio
          region:
            key: DEFAULT_REGION
            name: minio
          secretAccessKey:
            key: ACCESS_SECRET_KEY
            name: minio
      retentionPolicy: 30d
    bootstrap:
      recovery:
        recoveryTarget:
          targetTime: 2020-11-26 15:22:00.00000+00
        source: recovery-cluster
    externalClusters:
      - barmanObjectStore:
          destinationPath: s3://backups/
          endpointURL: http://minio:9000
          s3Credentials:
            accessKeyId:
              key: ACCESS_KEY_ID
              name: minio
            secretAccessKey:
              key: ACCESS_SECRET_KEY
              name: minio
          serverName: my-cluster-to-restore
        name: recovery-cluster
    imageName: ghcr.io/cloudnative-pg/postgis:15
    imagePullPolicy: IfNotPresent
    instances: 1
    maxSyncReplicas: 0
    minSyncReplicas: 0
    monitoring:
      enablePodMonitor: false
    postgresql:
      parameters: null
    storage:
      size: 8Gi
  3: |
    backupOwnerReference: self
    cluster:
      name: RELEASE-NAME-cnpg-cluster
    schedule: 1 2 3 * * 0
cluster with recovery enabled:
  1: |
    bootstrap:
      recovery:
        recoveryTarget:
          targetTime: 2020-11-26 15:22:00.00000+00
        source: recovery-cluster
    externalClusters:
      - barmanObjectStore:
          destinationPath: s3://backups/
          endpointURL: http://minio:9000
          s3Credentials:
            accessKeyId:
              key: ACCESS_KEY_ID
              name: minio
            secretAccessKey:
              key: ACCESS_SECRET_KEY
              name: minio
          serverName: my-cluster-to-restore
        name: recovery-cluster
    imageName: ghcr.io/cloudnative-pg/postgis:15
    imagePullPolicy: IfNotPresent
    instances: 1
    maxSyncReplicas: 0
    minSyncReplicas: 0
    monitoring:
      enablePodMonitor: false
    postgresql:
      parameters: null
    storage:
      size: 8Gi
cluster with scheduled backup enabled:
  1: |
    backup:
      barmanObjectStore:
        destinationPath: s3://backups/
        endpointURL: http://minio:9000
        s3Credentials:
          accessKeyId:
            key: ACCESS_KEY_ID
            name: minio
          region:
            key: DEFAULT_REGION
            name: minio
          secretAccessKey:
            key: ACCESS_SECRET_KEY
            name: minio
      retentionPolicy: 30d
    bootstrap:
      initdb:
        database: app
        owner: app
        postInitTemplateSQL: null
    imageName: ghcr.io/cloudnative-pg/postgis:15
    imagePullPolicy: IfNotPresent
    instances: 1
    maxSyncReplicas: 0
    minSyncReplicas: 0
    monitoring:
      enablePodMonitor: false
    postgresql:
      parameters: null
    storage:
      size: 8Gi
